### PR TITLE
Send CTA value for Link Popup

### PR DESCRIPTION
**Pull request by davidme-stripe**
## Summary
Send a CTA value to the Link popup. This will allow us to show "Set up" for SetupIntents.

## Testing
* Updated existing URL generation test (and sorted the keys to make it more deterministic).
* Also updated a test for invalid character encoding, as this seems to have been fixed in Xcode 15.3, so the behavior is inconsistent across versions. This test wasn't particularly useful anyway.
* Server-side change hasn't landed yet

## Changelog
Added

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/Link/LinkURLGeneratorTests.swift [hunk 1]
```diff
@@ -39,17 +41,6 @@ class LinkURLGeneratorTests: XCTestCase {
...
-        var params = testParams
-        params.customerInfo.email = String(bytes: [0xD8, 0x00] as [UInt8], encoding: .utf16BigEndian)! // Unpaired UTF-16 surrogates
-        do {
-            _ = try LinkURLGenerator.url(params: params)
-            XCTFail("Encoding should fail for invalid data")
-        } catch {
-            XCTAssertTrue(error is EncodingError)
-        }
-    }
-
```
Comment (wooj-stripe) [inline StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/Link/LinkURLGeneratorTests.swift:52]: We support horrible unicode now?

**Review by wooj-stripe — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,8 +1,10 @@
+## X.X.X
+### PaymentSheet
+* [Fixed] When confirming a SetupIntent with Link, "Set up" will be shown as the confirm button text instead of "Pay".
+
 ## 23.25.1 2024-03-18
 ### All
 * Xcode 14 is [no longer supported by Apple](https://developer.apple.com/news/upcoming-requirements/). Please upgrade to Xcode 15 or later.
-
-### PaymentSheet
 * [Fixed] A bug where `PaymentSheet.FlowController` was not respecting `PaymentSheet.Configuration.primaryButtonLabel`.
```
Comment (wooj-stripe) [inline CHANGELOG.md:8]: probably need to revert this.

File: StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/Link/LinkURLGeneratorTests.swift (hunk 1 above)
Comment (davidme-stripe) [reply StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/Link/LinkURLGeneratorTests.swift:52]: As of Xcode 15.3, but not Xcode 15.1. We could have two tests, but this test isn't really necessary IMO

File: CHANGELOG.md (hunk 2 above)
Comment (davidme-stripe) [reply CHANGELOG.md]: Ugh merging is hard

**Review by wooj-stripe — APPROVED** [review]

**Pull request merged**